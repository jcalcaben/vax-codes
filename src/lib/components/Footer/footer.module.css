.root {
	composes: text-gray-500 from global;
	composes: bg-gray-100 from global;
	composes: max-w-full from global;
	composes: grid from global;
	composes: justify-center from global;
	composes: px-12 from global;
	composes: py-6 from global;

	grid-template-columns: 3fr 1fr;
}

.link {
	composes: text-blue-600 from global;
	composes: underline from global;
}

.backToTop {
	composes: justify-self-end from global;
	composes: m-0 from global;
	composes: p-0 from global;
	composes: self-start from global;
	composes: col-span-2 from global;
	composes: col-start-2 from global;
	composes: row-start-1 from global;

	composes: link;
}

.message {
	composes: mb-6 from global;
}

.navigation {
	composes: flex from global;
	composes: space-x-12 from global;
}