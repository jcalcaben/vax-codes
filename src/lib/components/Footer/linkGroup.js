import React from "react";
import { Link } from "gatsby";

import { root, item, list, title } from "./linkGroup.module.css";

const LinkGroup = (props) => {
  const { data } = props;

  const { title: groupTitle, links } = data;

  const linkList = links.map((link) => {
    const { title: linkTitle, url } = link;

    const linkElement = url.startsWith("/") ? (
      <Link to={url}>{linkTitle}</Link>
    ) : (
      <a href={url} target="_blank" rel="noopener noreferrer">
        {linkTitle}
      </a>
    );

    return (
      <li className={item} key={linkTitle}>
        {linkElement}
      </li>
    );
  });

  return (
    <nav className={root}>
      <div className={title}>{groupTitle}:</div>
      <ul className={list}>{linkList}</ul>
    </nav>
  );
};

export default LinkGroup;
