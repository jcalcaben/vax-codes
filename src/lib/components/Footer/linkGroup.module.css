.root {

}

.item {
	composes: text-blue-600 from global;
	composes: underline from global;

}

.list {
	composes: list-none from global;
}

.title {
	composes: my-0.5 from global;
}
