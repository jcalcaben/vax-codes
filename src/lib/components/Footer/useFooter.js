import {
  helpfulLinks,
  qrCodes,
  adminPortal,
  otherLinks,
} from "../../data/footerLinks";

const useFooter = (props) => {
  return {
    helpfulLinks,
    qrCodes,
    adminPortal,
    otherLinks,
  };
};

export default useFooter;
