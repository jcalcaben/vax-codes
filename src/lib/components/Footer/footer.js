import React from "react";

import LinkGroup from "./linkGroup";
import useFooter from "./useFooter";

import {
  root,
  backToTop,
  message,
  navigation,
  link,
} from "./footer.module.css";

const Footer = (props) => {
  const { helpfulLinks, qrCodes, adminPortal, otherLinks } = useFooter(props);

  return (
    <footer className={root}>
      <p className={message}>
        Vax.Codes is an open-source project run by volunteers at&nbsp;
        <a
          className={link}
          href="https://www.open-austin.org/"
          target="_blank"
          rel="noopener noreferrer"
        >
          Open Austin
        </a>
        .
      </p>
      <div className={navigation}>
        <LinkGroup data={helpfulLinks} />
        <LinkGroup data={qrCodes} />
        <LinkGroup data={adminPortal} />
        <LinkGroup data={otherLinks} />
      </div>
      <a href="#top" className={backToTop}>
        Back to top
      </a>
    </footer>
  );
};

export default Footer;
