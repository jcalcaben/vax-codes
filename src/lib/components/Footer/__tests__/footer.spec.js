import React from 'react';
import Footer from '../footer';
import TestRenderer from 'react-test-renderer';

describe("Footer component", () => {
  test("renders the correct basic shape", () => {
    const tree = TestRenderer.create(<Footer/>);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
