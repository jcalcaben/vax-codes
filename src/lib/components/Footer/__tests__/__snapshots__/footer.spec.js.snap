// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders the correct basic shape 1`] = `
<footer
  className="root"
>
  <p
    className="message"
  >
    Vax.Codes is an open-source project run by volunteers atÂ 
    <a
      className="link"
      href="https://www.open-austin.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Open Austin
    </a>
    .
  </p>
  <div
    className="navigation"
  >
    <nav
      className="root"
    >
      <div
        className="title"
      >
        Helpful links
        :
      </div>
      <ul
        className="list"
      >
        <li
          className="item"
        >
          <a
            aria-current="page"
            className=""
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="item"
        >
          <a
            href="/how-it-works/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            How it works
          </a>
        </li>
        <li
          className="item"
        >
          <a
            href="/about/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            About us
          </a>
        </li>
      </ul>
    </nav>
    <nav
      className="root"
    >
      <div
        className="title"
      >
        QR Codes
        :
      </div>
      <ul
        className="list"
      >
        <li
          className="item"
        >
          <a
            href="/scan/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Scan a QR code
          </a>
        </li>
        <li
          className="item"
        >
          <a
            href="/issues/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Browse issuers
          </a>
        </li>
        <li
          className="item"
        >
          <a
            href="/groups/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Browse groups
          </a>
        </li>
      </ul>
    </nav>
    <nav
      className="root"
    >
      <div
        className="title"
      >
        Admin portal
        :
      </div>
      <ul
        className="list"
      >
        <li
          className="item"
        >
          <a
            href="/admin/issuers/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Issuers admin
          </a>
        </li>
        <li
          className="item"
        >
          <a
            href="/admin/groups/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Group owners
          </a>
        </li>
      </ul>
    </nav>
    <nav
      className="root"
    >
      <div
        className="title"
      >
        Other links
        :
      </div>
      <ul
        className="list"
      >
        <li
          className="item"
        >
          <a
            href="/security/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Security
          </a>
        </li>
        <li
          className="item"
        >
          <a
            href="/developers/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            API docs
          </a>
        </li>
        <li
          className="item"
        >
          <a
            href="https://github.com/open-austin/vax-codes"
            rel="noopener noreferrer"
            target="_blank"
          >
            Github repo 
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <a
    className="backToTop"
    href="#top"
  >
    Back to top
  </a>
</footer>
`;
